packages/@lwc/synthetic-shadow/src/shared/event-target.ts

```diff
@@ -47,9 +57,8 @@ export function getEventListenerWrapper(
             }
 
             return isFunction(fnOrObj)
-                ? (fnOrObj as EventListener).call(this, event)
-                : (fnOrObj as EventListenerObject).handleEvent &&
-                      (fnOrObj as EventListenerObject).handleEvent(event);
+                ? fnOrObj.call(this, event)
+                : fnOrObj.handleEvent && fnOrObj.handleEvent(event);
```

Comment:
```suggestion
                : fnOrObj.handleEvent?.(event);
```

Reply:
Didn't we revert ES2020 support?

Reply:
This is done by the TypeScript compiler so it should be fine.